test/optimizations/deadCodeElimination/elliot/countDeadModules.good

```diff
@@ -1 +1 @@
-Removed 26 dead modules.
+Removed 27 dead modules.
```

Comment:
This change surprised me (in the "aren't we just substituting one module for another sense") until I realized it's because internal (and maybe standard?) modules still use `Math`.  Did you happen to investigate whether those uses of Math could be removed?  It seems as though if our separation into AutoMath vs. Math reflects commonly-used things vs. not, we might be the case that internal/standard modules would only need AutoMath, which I think would be a win.  Removing the `use Math;` in `ChapelRange.chpl` for example, some simple tests that exercise ranges, domains, and arrays still work.

Reply:
I chalked it up to one of the symbols in the AutoMath module using `logBasePow2` under the covers, so didn't look too deeply into it.  I think it's more worth looking into once we've finalized how things will be sorted

Reply:
My hope would also be that internal/standard modules would mostly just need AutoMath, though

Reply:
Oh, I hadn't noticed that `AutoMath` itself uses `Math`.  That's unfortunate.  :(

Reply:
It doesn't bleed the symbols from it into other scopes, so it is still an improvement.  I can't think of another way to organize it that would both:
- not use `Math` and
- not make `logBasePow2` included into the default namespace
We could reconsider making `logBasePow2` an auto-included symbol, but it didn't seem like anyone wanted that for its own sake.

Reply:
I definitely agree it's an improvement, and am glad for the PR.  But better would be for every program not to have to parse all of the `Math` module, particularly as it gets larger and `AutoMath` gets smaller.  I'd suggest making an internal / developer-only version of `logBasePow2`, having `AutoMath` call that rather than using `Math` and then having `Math`'s `logBasePow2` call the developer routine.

Reply:
(though we'd still need to remove other uses of Math from internal/standard modules, so this would be a step in the right direction, but wouldn't get us all the way there).

Reply:
We can do that, but if we mark it "private", Math can't reference it, and if we don't, it'll bleed into the default namespace.  We could use the `chpl_` prefix to make it less likely to conflict with user symbols, though . . .

Reply:
I could also make a helper module that AutoMath and Math use but don't re-export?  That'd solve the symbol bleed problem

Reply:
> We could use the chpl_ prefix to make it less likely to conflict with user symbols, though . . .

This is what I was imagining.  Introducing another module would also work, though it feels like overkill to me here.

Reply:
Oh, sorry, also forgot that the deprecated versions of `log1p` and `logBasePow2` in AutoMath call to their Math implementation under the covers (to minimize the potential for code drift).  It seems a little finicky to have the implementation of all the moved functions stay in AutoMath until the deprecation warnings have been removed.  But the deprecated versions will be shorter-lived than the symbols that refer to `logBasePow2` otherwise, so I think the original proposed change is still worth doing in the meanwhile.  Does that sound right to you, Brad?

Reply:
Yeah, I agree it's a little entangled, but also think it's the right way to go.